src/dataflow-types/src/types.rs

```diff
@@ -59,8 +59,10 @@ impl PeekResponse {
 /// Various responses that can be communicated about the progress of a TAIL command.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub enum TailResponse {
+    /// Progress information about the upper frontier.
+    Progress(Timestamp),
```

Comment:
Would you mind using an `Antichain<Timestamp>` here? Elsewhere in `dataflow` we've tried to stay away from assumptions of total order / don't want to head more in that direction.

Reply:
Also, if you would flesh out the comment to indicate *what* progress this implies (e.g. that subsequent messages from this worker will only contain timestamps greater or equal to this time).

Reply:
Done!